tests/modeltests/tst_disassemblyoutput.cpp

```diff
@@ -207,30 +207,32 @@ private slots:
         QVERIFY(result.errorMessage.isEmpty());
 
         auto isValidVisualisationCharacter = [](QChar character) {
-            const static auto validCharacters =
-                std::initializer_list<QChar> {QLatin1Char(' '),  QLatin1Char('\t'), QLatin1Char('|'), QLatin1Char('/'),
-                                              QLatin1Char('\\'), QLatin1Char('-'),  QLatin1Char('>'), QLatin1Char('+')};
+            const static auto validCharacters = std::initializer_list<QChar> {
+                QLatin1Char(' '), QLatin1Char('\t'), QLatin1Char('|'), QLatin1Char('/'), QLatin1Char('\\'),
+                QLatin1Char('-'), QLatin1Char('>'),  QLatin1Char('+'), QLatin1Char('X')};
 
             return std::any_of(validCharacters.begin(), validCharacters.end(),
                                [character](auto validCharacter) { return character == validCharacter; });
         };
 
+        auto isValidHexdumpCharacter = [](QChar character) {
+            const static auto validCharacters = std::initializer_list<QChar> {
+                QLatin1Char(' '), QLatin1Char('0'), QLatin1Char('1'), QLatin1Char('2'), QLatin1Char('3'),
+                QLatin1Char('4'), QLatin1Char('5'), QLatin1Char('6'), QLatin1Char('7'), QLatin1Char('8'),
+                QLatin1Char('9'), QLatin1Char('a'), QLatin1Char('b'), QLatin1Char('c'), QLatin1Char('d'),
+                QLatin1Char('e'), QLatin1Char('f')};
+            return std::any_of(validCharacters.begin(), validCharacters.end(),
+                               [character](auto validCharacter) { return character == validCharacter; });
+        };
+
         for (const auto& line : result.disassemblyLines) {
             QVERIFY(!line.branchVisualisation.isEmpty());
 
             // check that we only captures valid visualisation characters
             QVERIFY(std::all_of(line.branchVisualisation.cbegin(), line.branchVisualisation.cend(),
                                 isValidVisualisationCharacter));
 
-            QVERIFY(!line.disassembly.isEmpty());
```

Comment:
why did you remove the old code that checks that the disassembly wasn't empty? that should still be the case, no?

Reply:
no, sometimes we encounter `nop`s like this:
```asm
    124a:       f2 0f 10 4c 24 08       movsd  0x8(%rsp),%xmm1
    1250:       f2 0f 10 44 24 10       movsd  0x10(%rsp),%xmm0
    1256:       f2 0f 58 0d aa 0d 00    addsd  0xdaa(%rip),%xmm1        # 2008 <_IO_stdin_used+0x8>
    125d:       00 
    125e:       ff c3                   inc    %ebx
    1260:       81 fb 40 42 0f 00       cmp    $0xf4240,%ebx
    1266:       0f 85 64 ff ff ff       jne    11d0 <main+0x50>
```

Reply:
That's important information, maybe add here as comment?

Reply:
I agree, add a comment - better yet explicitly catch it - i.e. expect empty when the hexdump is `00`, otherwise expect non-empty